content/projects/gsoc/2023/project-ideas/gitlab-plugin-modernization.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: gsocprojectidea
+title: "GitLab Plugin Modernization"
+goal: "Cleaning and modernizing the extensively used GitLab plugin"
+category: Plugin improvement
+year: 2023
+status: draft
+sig: platform
+skills:
+- Java
+- Docker
+- GitLab
+mentors:
+- "gsoc"
+
+links:
+    emailThread: https://community.jenkins.io/t/gsoc-2023-project-idea-active-modernization-of-gitlab-plugin/5039
+//   gitter: "jenkinsci/plugin-installation-manager-cli-tool"
+//   draft: https://docs.google.com/document/d/1s-dLUfU1OK-88bCj-GKaNuFfJQlQNLTWtacKkVMVmHc
+---
+=== Background
+
+The GitLab plugin is widely used (58,727 installations) but has not had an active maintainer for a long time. 
+Many of us at CloudBees have volunteered to keep it afloat over the years, including Owen, Basil, Bruno, Jean-Marc, and Mark. 
+But it could use a sprint of active development. 
+Maybe GSoC could be a good way to staff this with an intern?
+
+At first sight, there are a number of things that need to be done to keep this plugin afloat in the long term:
+
+* Migrate from the very old version of RESTEasy it is using to the GitLab API plugin (just like GitLab Authentication, GitLab Branch Source, and GitLab Logo)
+* Revive the Docker-based functional test suite. It has been broken for some time and is currently disabled.
+* Fix any of the dozens of outstanding bugs.
+
+The objectives could be expanded to cover the "sister" `gitlab-branch-source` plugin.
+
+This would be a good GSoC project for the following reasons:
+
+* While boasting less of a brag factor than the link:/projects/gsoc/2023/project-ideas/agent_reconnections_exponential_backoff/[exponential backoff and jitter idea], impact to a large number of users might be brag factor enough.
+* Would be ideal for someone who wants to get GitLab experience and get paid along the way.
+* Feasible within three months: Nothing prohibitively hard here, just a matter of doing the work (cutting code and testing).
+* Stretchable and shrinkable scope: If the intern is struggling, we can drop one of the two main changes. 
+If the intern is doing better than expected, we can fix more bugs.
+* Not fatal if the project fails: People still seem to keep using this plugin no matter how poorly maintained it is.
+
+// === Quick Start
+// TBD
+//
+=== Skills to Study and Improve
+
+- Java
+- Docker
+- GitLab
+
+=== Project Difficulty Level
+
+Beginner to Intermediate
+
+=== Project Size
+
+175 hours
```

Comment:
The description describes three different specific topics …

1. RESTeasy
2. Functional test suite
3. Bug fixing

… as well as a fourth non-specific topic:

4. `gitlab-branch-source-plugin`

All four of these would definitely stretch to a 350 hour project IMHO. That was my thinking when I wrote the project description. Why was I not consulted before this PR was merged?

If this project is to remain as a 175 hour project, I think a subset of the above should be selected, or this project should be split into multiple 175 hour projects.

Reply:
reverted back to "discussion" via #5964 
See thread at https://community.jenkins.io/t/gsoc-2023-project-idea-active-modernization-of-gitlab-plugin/5039